### ISSUE TITLE
Prolog Lexer

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#152]
(Original issue 373 created by None on 2008-11-12T16:22:09+00:00)

It would be nice to have prolog syntax highlight

Reported by guest

**Comment by Anteru** [comment]
(Original issue was assigned to thatch)

(Original issue 373 last updated on 2009-01-03T16:42:48+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by thatch on 2009-01-03T16:42:48+00:00)

Yes, this is now in pygments-main.  See [http://dev.pocoo.org/projects/pygments/browser/pygments/lexers/compiled.py#L1342]... with #355 now fixed, it does some quick detection to tell perl from prolog files.  This is done in `analyse_text`, which currently looks for `:-` but I intend to add detection of `/^%/` for comments that start a line.  Feel free to suggest better heuristics.

**Comment by Anteru** [comment]
(Original comment by None on 2009-01-03T16:15:04+00:00)

good work. is this in head already? Using .pl is problematic but other suffixes used are .pro and .prolog

**Comment by Anteru** [comment]
(Original comment by thatch on 2008-12-31T03:37:10+00:00)

I agree, it should be in here.  I've written an initial version after consulting with a friend, which is around 95% complete on the syntax (I think there are some finicky bits with operators, and would appreciate some review).  It's currently in [http://code.timhatch.com/hg/pygments-tim my branch], and depends on #355 before `.pl` can be added to the filenames.

**Comment by Anteru** [comment]
(Original comment by gbrandl on 2008-11-15T11:11:22+00:00)

I agree :)

Let's see if someone writes one; I don't know Prolog well enough to do it off the cuff.
